include/experimental/group/gemm/impl/int4_dequantize_xmx_xe.hpp

```diff
@@ -389,7 +399,9 @@ class gemm_t<
             if constexpr (enable_periodic_sync) {
                 if ((i % sync_freq) == 0) {
                     if constexpr (wg_size_x > 1) { nbarrier_a.arrive(); }
-                    if constexpr (wg_size_y > 1) { nbarrier_b.arrive(); }
+                    if constexpr (arch_tag >= gpu_arch::Xe) {
+                        if constexpr (wg_size_y > 1) { nbarrier_b.arrive(); }
```

Comment:
any benefit here? It seems only two barriers' sync points, and no other interactive codes after sync.

Reply:
Set 2 synchronization points in PVC. Here is the historical code. The instruction can correspond to producer_consumer, and there is no similar instruction in Dg2. Keep only one barrier